### PR TITLE
[WIP] `eslint` code hinting

### PR DESCRIPTION
**Pull request by toxamiz**
I've tried to configure eslint rules, based on `scripts/helper.js`. Can use this for more accurate further settings.

### CODE REVIEW

**Comment by adamzerella** [comment]
Thanks @toxamiz! I'll review the linting rules over the next few days, looks like a really good start though. We should also be looking at tools like https://www.npmjs.com/package/husky for when commits are made and ideally everything is formatted automatically.

**Comment by toxamiz** [comment]
For first look there are some indent issues (or the base file have indent issues :) ) and there no base rule for UpperCamesCase for global functions ` const CreateDir = ( dir ) => {...} `. 
Eslint can be extended with custom rules, if there are few missing rules, we can try to make them (or try to find)

**Comment by adamzerella** [comment]
Looks good @toxamiz, some more feedback:

- Rename `eslintconfigs/` to `.eslint/`
- Add a lint script for jQuery, something like: `"lint:js": "eslint -c .eslint/jquery.json packages/*/src/js/jquery.js"`

We need to make a conscious effort that the `module.js` and `jquery.js` files are IE8+ compliant. 

Looking at a plugin like: https://www.npmjs.com/package/eslint-plugin-ie11 might be a suitable option, or at least inheriting some of the rules from this.

File: eslintconfigs/module.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+	'env': {
+		'browser': true
+	},
+	'parserOptions': {
+		'ecmaVersion': 5
```
Comment (adamzerella) [inline eslintconfigs/module.js]:
```suggestion
		'ecmaVersion': 5
```

I'm not confident this supports IE8-11 rules

Comment (toxamiz) [reply eslintconfigs/module.js]: It just forbid to use es6 syntax e.g. `let`, `const`, etc

**Comment by toxamiz** [comment]
> https://www.npmjs.com/package/eslint-plugin-ie11 might be a suitable option, or at least inheriting some of the rules from this.

I don't think it is what we need

I tried to use https://github.com/amilajack/eslint-plugin-compat which should lint the browser compatibility, but i doesn't work properly: allowing to use [].map on ie8 for example.

Btw, don't you think about to just require `module.js` and `jquery.js` to be written only with es5 features (which is supported by ie9+) or use babel transpile, and advise your component users to include [es5-shim](https://github.com/es-shims/es5-shiml) and [DOM-shim](https://github.com/Raynos/DOM-shim) if they need support of ie8?.

With es5 requirement we can use  `'env': {'es6': false}, 'parserOptions': {'ecmaVersion': 5}`

**Comment by adamzerella** [comment]
Closing this in favour of #710. We'll move this discussion over to #710 @toxamiz

**Pull request closed without merging**